Updated project to be compatible with recent Flutter master

KevinTheGray opened the pull request:
This PR updates the repo to be compatible with a very recent version of Flutter, 1.10.6-pre.40.  

Below, when talking about the "current Flutter example", I am referencing the example currently generated on the master channel by `flutter create --macos`

- Updated `README` to specify that the master channel must be used, and that it may be unstable.
- Updated `main.dart` to more closely resemble current Flutter example
- Updated the main `NSWindow` file to more closely resemble the current Flutter example
- Updated `MainMenu.xib` to more closely resemble current Flutter example.  Removes references to many FDE resources that were in there before
- Renamed one of the targets to `Flutter Assemble`, and updated that target to match the current Flutter example
- Re-added Flutter Icons
- Removed a reference to a non-existing path in the pubspec.yaml
- Updated plugin git references to more recent versions
- git ignoring .dart_tool directory
- Added the Podfile and Podfile.lock files

KevinTheGray commented:
@stuartmorgan @timsneath, here's the PR I was referencing earlier.  

A good chunk of these changes are updates to match the example provided by Flutter.

stuartmorgan commented on README.md:
```diff
@@ -20,8 +20,8 @@ For more details on the [presentation file structure](https://github.com/flutter
 ### Requirements
-- Flutter version `v1.7.8+hotfix.3`, but not on the `stable` channel. Older or newer versions of Flutter are likely to be incompatible. Run `flutter version 1.7.8+hotfix.3` to check out a compatible version.
-- Ensure that the Xcode section of `flutter doctor` does not list any issues with the version of Flutter above.
+- Flutter on the `master` channel. FDE is still not stable, so you may need to do some tweaking to get it running.  The last verified working Flutter version is `1.10.6-pre.40` on 09/24/2019.
```
Remove reference to FDE; it's not involved at all at this point.

I think you could remove everything after the first sentence at this point, in fact. I have no expectation that this will break going forward.

KevinTheGray replied:
How's the update look?  Went with removing everything after the first sentence

stuartmorgan replied:
Looks good. (I noticed while looking at the full file that the intro still says it's using Flutter Desktop Embedding; I'd suggest changing that to say something like "Flutter's in-progress support for Desktop", with the link going to https://flutter.dev/desktop)

KevinTheGray replied:
Updated this as suggested

stuartmorgan commented on flutter_app/macos/Assets.xcassets/AppIcon.appiconset/Contents.json:
```diff
@@ -3,61 +3,61 @@
     {
       "size" : "16x16",
       "idiom" : "mac",
-      "filename" : "16pt.png",
```
I would expect to see file moves for these old icons, but that's not in the diff. Are there two copies of all the icons now?

KevinTheGray replied:
These were removed in the last PR, probably a copy/paste issue.   I've also double checked with a `find . -name "16pt.png"` with no results, so I think this one is resolved.

KevinTheGray commented:
I just noticed the Wiki has been removed, which previously had details on how to structure a presentation file.  Do we want to re-add it, or was it removed intentionally?  If there is a preferred way to list that type of documentation, I could re-add in the information in that way.

stuartmorgan commented:
> I just noticed the Wiki has been removed, which previously had details on how to structure a presentation file. Do we want to re-add it, or was it removed intentionally?

I don't know anything about how the repository itself has been managed; maybe @timsneath knows what happened there.

timsneath commented:
Strange -- no, I don't know anything about what happened there...

On Tue, Sep 24, 2019 at 10:47 AM stuartmorgan <notifications@github.com>
wrote:

> I just noticed the Wiki has been removed, which previously had details on
> how to structure a presentation file. Do we want to re-add it, or was it
> removed intentionally?
>
> I don't know anything about how the repository itself has been managed;
> maybe @timsneath <https://github.com/timsneath> knows what happened there.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/flutter/slideplayer/pull/9?email_source=notifications&email_token=AARWL64PQFLSJU4RI2LIZSLQLJHDNA5CNFSM4I2CO4FKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7PG72A#issuecomment-534671336>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARWL656YYWHT4BTLMWDJ7LQLJHDNANCNFSM4I2CO4FA>
> .
>

KevinTheGray commented:
@stuartmorgan I added the minimum flutter version, and I also regenerated the flutter app using `flutter create`.  I did have to remove the App Sandbox entitlements due to how the app is accessing and listening to files, so it should more closely resemble the example app.   

I also updated the README as we discusses above.

As for the wiki, I suggest I re-add it as a separate PR.

stuartmorgan commented on flutter_app/macos/Runner/Configs/AppInfo.xcconfig:
```diff
@@ -0,0 +1,14 @@
+// 'flutter create' template.
+
+// The application's name. By default this is also the title of the Flutter window.
+PRODUCT_NAME = flutter_app
```
This should be "Flutter Slides"

KevinTheGray replied:
Sounds good, will fix

KevinTheGray replied:
This is now updated to Flutter Slides

stuartmorgan commented on flutter_app/macos/Runner/Configs/AppInfo.xcconfig:
```diff
@@ -0,0 +1,14 @@
+PRODUCT_NAME = flutter_app
+
+// The application's bundle identifier
+PRODUCT_BUNDLE_IDENTIFIER = com.example.flutterApp
```
And this and the copyright should be whatever they were in the old files.

KevinTheGray replied:
so, the old bundle ID was `com.vgventures.flutterslides`, should I change this to `com.google.flutterslides`?   Also, does google have  preferred product copyright I should put in there? The old copyright just said `Copyright © 2018 VGVentures`.

stuartmorgan replied:
@timsneath Can you offer guidance here? For bundle ID we generally use io.flutter.<whatever> I believe, but I'm not sure how Copyright should be handled in this case.

timsneath replied:
`io.flutter` would be good for the bundle. 

For the license, please use `Copyright (C) 2017 The Chromium Authors`. 

And also, please grab https://github.com/flutter/devtools/blob/master/LICENSE and put it in the root.

KevinTheGray replied:
Set that bundle id and copyright as suggested and added the license

stuartmorgan commented on flutter_app/macos/Runner/Info.plist:
```diff
@@ -7,7 +7,7 @@
 	<string>$(EXECUTABLE_NAME)</string>
 	<key>CFBundleIconFile</key>
-	<string>AppIcon.appiconset</string>
+	<string></string>
```
Why is this removed?

KevinTheGray replied:
It looks like this isn't the default way to specify an app icon, and apparently CFBundleIcon is now a legacy way to set the icons according to this page: https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html

When we `flutter create` a new macOS app, it generates and sets ASSET_CATALOG_APPICON_NAME for each configuration.  I've verified that the app icon is being used and show when you run the app.

stuartmorgan replied:
Ah right, I forgot about that from doing the icon change in the example. Sorry for the noise.

KevinTheGray commented:
@stuartmorgan @timsneath if those comments are resolved, I think we're safe to merge this one in.

timsneath commented:
Lgtm

On Fri, Sep 27, 2019, 1:31 PM KevinTheGray <notifications@github.com> wrote:

> @stuartmorgan <https://github.com/stuartmorgan> @timsneath
> <https://github.com/timsneath> if those comments are resolved, I think
> we're safe to merge this one in.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/flutter/slideplayer/pull/9?email_source=notifications&email_token=AARWL6ZRJI3NKJHIW7DZCCDQLZURZA5CNFSM4I2CO4FKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD72AQAA#issuecomment-536086528>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARWL66OLIURYVNDNOHO3XTQLZURZANCNFSM4I2CO4FA>
> .
>

stuartmorgan commented:
@KevinTheGray Something that would be nice to do as a follow-up would be to remove the custom recent-files method channel and replace it with shared_preferences (+shared_preferences_fde for the macOS implementations). Then this could easily be run on Windows and Linux once they support that standard plugin, instead of needing custom platform-specific logic.

KevinTheGray commented:
@stuartmorgan good idea, I'll look into that
